pkg/cloud/aws/services/ec2/subnets.go

```diff
@@ -175,14 +175,56 @@ func (s *Service) describeVpcSubnets() (v1alpha1.Subnets, error) {
 		return nil, errors.Wrapf(err, "failed to describe subnets in vpc %q", s.scope.VPC().ID)
 	}
 
+	routeTables, err := s.describeVpcRouteTablesBySubnet()
+	if err != nil {
+		return nil, err
+	}
+
+	natGateways, err := s.describeNatGatewaysBySubnet()
+	if err != nil {
+		return nil, err
+	}
+	internetGateways, err := s.describeVpcInternetGateways()
+	if err != nil {
+		return nil, err
+	}
+
 	subnets := make([]*v1alpha1.SubnetSpec, 0, len(out.Subnets))
 	for _, ec2sn := range out.Subnets {
+		rt := routeTables[*ec2sn.SubnetId]
+		var hasPublicRoute bool
+		var routeTableId, natGatewayId *string
+		if rt != nil {
```

Comment:
Complexity can be reduced a little:

```
rt, ok := routeTables[*ec2sn.SubnetId]
if !ok {
    continue
}
```

which removes the need for this`if` block.

Reply:
I ended up reverting this change to simplify the tests (the `continue` skipped over the `append` below), and because it felt a little more natural with the tag stuff to have each block as a "fact" that we were trying to discover.